design/cluster-update.md

```diff
@@ -0,0 +1,209 @@
+#### Supported Fields
+The following fields will be **supported** for updates:
+* `monCount`: The number of monitors can be updated and the operator will ensure that as monitors are scaled up or down the cluster remains in quorum.
+* `deviceFilter`: The regex filter for devices allowed to be used for storage can be updated and OSDs will be added or removed to match the new filter pattern.
+* `useAllDevices`: If this value is updated to `true`, then OSDs will be added to start using all devices on nodes.
+However, if this value is updated to `false`, the operator will only allow OSDs to be removed if there is a value set for `deviceFilter`.
+This is to prevent an unintentional action by the user that would effectively remove all data in the cluster.
+* `useAllNodes`: This value will be treated similarly to `useAllDevices`.
+Updating it to `true` is a safe action as it will add more nodes and their storage to the cluster, but updating it to `false` is not always a safe action.
+If there are no individual nodes listed under the `nodes` field, then updating this field to `false` will not be allowed.
+* `resources`: The CPU and memory limits can be dynamically updated.
+* `placement`: The placement of daemons across the cluster can be updated, but it is dependent on the specific daemon.
+For example, monitors can dynamically update their placement as part of their ongoing health checks.
+OSDs can not update their placement at all since they have data gravity that is tied to specific nodes.
+Other daemons can decide when and how to update their placement, for example doing nothing for current pods and only honoring new placement settings for new pods.
+* `nodes`: Specific storage nodes can be added and removed, as well as additional properties on the individual nodes that have not already been described above:
+  * `devices`:  The list of devices to use for storage can have entries added and removed.
+  * `directories`: The list of directories to use for storage can also be updated.
+
+#### Unsupported Fields
+All other properties not listed above are **not supported** for runtime updates.
+Some particular unsupported fields to note:
+* `dataDirHostPath`: Once the local host directory for storing cluster metadata and config has been set and populated, migrating it to a new location is not supported.
+* `hostNetwork`: After the cluster has been initialized to either use host networking or pod networking, the value can not be changed.
+Changing this value dynamically would very likely cause a difficult to support transition period while pods are transferring between networks and would certainly impact cluster health.
+* `versionTag`: This field is likely to be removed to simplify versioning between Rook system and cluster namespaces, see [issue 1215](https://github.com/rook/rook/issues/1215) for more details.
```

Comment:
The field has been removed so we can remove it here too. :)

Reply:
Good catch, I will remove it.